lib/application.js

```diff
@@ -609,13 +610,16 @@ app.render = function render(name, options, callback) {
  *    http.createServer(app).listen(80);
  *    https.createServer({ ... }, app).listen(443);
  *
- * @return {http.Server}
+ * @return {Promise<http.Server>}
  * @public
  */
 
 app.listen = function listen() {
   var server = http.createServer(this);
-  return server.listen.apply(server, arguments);
+  return promisify(server.listen).apply(server, arguments)
+    .then(function () {
+      return server
+    });
 };
```

Comment:
@wesleytodd done

This could be written without util.promify like 
```js
  var args = Array.prototype.filter.call(arguments, function(arg){return typeof arg !=='function'});

  return new Promise(function (resolve, reject) {
    server.listen.apply(server, args.concat(function (err) {
      if (err) return reject(err);
      resolve(server);
    }));
  });
```